service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -168,6 +175,56 @@ private void checkAdminOrSiteAdopter(JWTData userData, int siteId) throws AuthEx
     }
   }
 
+  /**
+   * Check if the user is an admin or the uploader of the tree image with the given imageId
+   *
+   * @param userData the user's data
+   * @param imageId the ID of the site image to check
+   * @throws AuthException if the user is not an admin or the image's uploader
+   */
+  private void checkAdminOrImageUploader(JWTData userData, int imageId) throws AuthException {
+    if (isAdmin(userData.getPrivilegeLevel())) {
+      return;
+    }
+
+    int uploaderId =
+        db.select(SITE_IMAGES.UPLOADER_ID)
+            .from(SITE_IMAGES)
+            .where(SITE_IMAGES.ID.eq(imageId))
+            .fetchOne(0, int.class);
+    if (uploaderId != userData.getUserId()) {
+      throw new AuthException("User needs to be an admin or the image's uploader.");
+    }
+  }
+
+  /**
+   * Check if the user is able to upload a tree image. A user can upload if they are an admin or
+   * have less than `MAX_SUBMITTED_SITE_IMAGES` uploaded images that are waiting for admin approval
+   *
+   * @param userData the user's data
+   * @throws AuthException if the user has uploaded
+   */
+  private void checkCanUploadImage(JWTData userData) throws AuthException {
+    if (isAdmin(userData.getPrivilegeLevel())) {
+      return;
+    }
+
+    int numSubmittedImages =
+        db.selectCount()
+            .from(SITE_IMAGES)
+            .where(SITE_IMAGES.UPLOADER_ID.eq(userData.getUserId()))
+            .and(SITE_IMAGES.APPROVAL_STATUS.eq(String.valueOf(ImageApprovalStatus.SUBMITTED)))
+            .fetchOne(0, int.class);
+
+    if (numSubmittedImages > MAX_SUBMITTED_SITE_IMAGES) {
+      throw new ForbiddenException(
+          String.format(
```

Comment:
⛏️ Small nitpick, can we change the `>` to `>=`? This check won't throw if the user has exactly the max number of of submitted images, allowing them to technically submit one more than the max.